File: top-interview-questions-medium/Sorting and Searching/find-peak-element/162.py
class Solution:
    def findPeakElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        min = 0
        max = len(nums) - 1
        mid = int(max / 2)
        while min < max:
            if min == mid: return max if nums[min] < nums[max] else min
            min = mid if nums[mid] < nums[mid + 1] else min
            max = mid if nums[mid] > nums[mid + 1] else max
            mid = int((min + max) / 2)

        return 0

File: top-interview-questions-medium/Sorting and Searching/find-peak-element/162.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findPeakElement = function(nums) {
  let min = 0
  let max = nums.length - 1
  for (let mid = Math.floor(max / 2); min < max; mid = Math.floor((min + max) / 2)) {
    if (min === mid) return nums[min] < nums[max] ? max : min
    min = nums[mid] < nums[mid + 1] ? mid : min
    max = nums[mid] > nums[mid + 1] ? mid : max
  }
  return 0
}
